src/main/java/com/team4/testingsystem/services/impl/TestGeneratingServiceImpl.java

```diff
@@ -43,19 +45,31 @@ public Test formTest(Test test) {
         setQuestionsByModules(test, Modules.SPEAKING.getName(), Pageable.ofSize(oneElement));
         return test;
     }
-    
+
     private void setQuestionsByModules(Test test, String module, Pageable pageable) {
         Collection<Question> questions = questionRepository
                 .getRandomQuestions(test.getLevel().getName(), module, pageable);
+
+        if (questions.size() < pageable.getPageSize()) {
+            throw new NotEnoughQuestionsException("Not enough questions for " + module + " " + test.getLevel());
+        }
+
         questions.forEach(test::setQuestion);
     }
 
     private void setQuestionsByContentFile(Test test) {
-        ContentFile contentFile = contentFilesService
-                .getRandomContentFile(test.getLevel().getName());
-        Collection<Question> questions = questionRepository
-                .getRandomQuestionByContentFile(contentFile.getId(), Pageable.ofSize(count));
-        questions.forEach(test::setQuestion);
+
+        try {
+            ContentFile contentFile = contentFilesService
+                    .getRandomContentFile(test.getLevel().getName());
+
+            Collection<Question> questions = questionRepository
+                    .getRandomQuestionByContentFile(contentFile.getId(), Pageable.ofSize(count));
+
+            questions.forEach(test::setQuestion);
```

Comment:
Is it work like addQuestion? Set question seams a bit strange here. Anyway, can we add all the questions instead adding them one by one?